### PR TITLE
set default language based on browser language

### PR DESCRIPTION
**Pull request by henryzbw**
detect browser language first, then check localstorage, if localstorage doesn't exist, set the default language by browser language, if the user changed the language once, it will set localstorage,  the new language will become the default language next time he starts the application.

### CODE REVIEW

File: src/appProvider.js [hunk 1]
```diff
@@ -11,13 +11,20 @@ import AppLocale from './languageProvider';
 import bodhiTheme from './config/theme';
 import graphClient from './network/graphClient';
 import { store, history } from './redux/store';
-
 import '../src/style/styles.less';
 
+const language = (navigator.languages && navigator.languages[0]) ||
```
Comment (frankobe) [inline src/appProvider.js]: consider match elements in `navigator.languages` in order.

File: src/appProvider.js [hunk 2]
```diff
@@ -11,13 +11,20 @@ import AppLocale from './languageProvider';
...
+                     navigator.language ||
+                     navigator.userLanguage;
+let browserLang = 'mandarin';
```
Comment (frankobe) [inline src/appProvider.js]: why not using `AppLocale.zh`?

File: src/appProvider.js [hunk 3]
```diff
@@ -11,13 +11,20 @@ import AppLocale from './languageProvider';
...
+if (language.includes('en')) {
```
Comment (frankobe) [inline src/appProvider.js]: Need ensure language starts with `en` or `en-`.  `en` can be in the middle of the string

File: src/appProvider.js (hunk 2 above)
Comment (alex-cory) [reply src/appProvider.js]: readability. Since we're only using 2 languages right now, I implemented it this way

Comment (frankobe) [reply src/appProvider.js]: @alex-cory dunt think readability is an issue, `cn` and `en` are commonly used

File: src/appProvider.js (hunk 1 above)
Comment (alex-cory) [inline src/appProvider.js]: this code block should go in `componentDidMount`

File: src/appProvider.js (hunk 2 above)
Comment (alex-cory) [reply src/appProvider.js]: yeah I worked with him to get a better solution :)

**Pull request merged**